### PR TITLE
Add ES Search IgnoreUnavailable indices flag

### PR DESCRIPTION
**Pull request by mh-park**
https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-index.html

### CODE REVIEW

File: plugin/storage/integration/fixtures/queries.json [hunk 1]
```diff
@@ -81,8 +81,8 @@
       "ServiceName": "query05-service",
       "OperationName": "",
       "Tags": null,
-      "StartTimeMin": "2017-01-26T15:46:31.639875Z",
-      "StartTimeMax": "2017-01-26T17:46:31.639875Z",
+      "StartTimeMin": "2017-01-26T00:00:31.639875Z",
```
Comment (black-adder) [inline plugin/storage/integration/fixtures/queries.json]: why are these changes required?

File: plugin/storage/es/dependencystore/storage.go [hunk 2]
```diff
@@ -121,17 +122,12 @@ func buildTSQuery(endTs time.Time, lookback time.Duration) elastic.Query {
 	return elastic.NewRangeQuery("timestamp").Gte(endTs.Add(-lookback)).Lte(endTs)
 }
 
-func (s *DependencyStore) getIndices(ts time.Time, lookback time.Duration) []string {
+func getIndices(ts time.Time, lookback time.Duration) []string {
 	var indices []string
 	indices = append(indices, indexName(ts))
 	for lookback > 0 {
```
Comment (black-adder) [inline plugin/storage/es/dependencystore/storage.go]: didn't you have some logic in a different file that did this more carefully? ie even if ts - lookback falls on the same day as ts, we're still adding the index for the previous day,

Comment (mh-park) [inline plugin/storage/es/dependencystore/storage.go]: hmm no that wasn't ever my intention.. but now that i have the ignoreunavailable flag, that could be safer

File: plugin/storage/integration/fixtures/queries.json (hunk 1 above)
Comment (mh-park) [inline plugin/storage/integration/fixtures/queries.json]: so the spans in the fixture were too far apart (i.e. ~ a day) so the reader wasn't able to retrieve it, because i made the logic in the reader based on that traces won't be more than ~2 hours long (idk if you remember talking about this with me a few weeks back)

Comment (black-adder) [inline plugin/storage/integration/fixtures/queries.json]: I don't even remember what I had for breakfast, can you point me to the code in question?

File: plugin/storage/es/dependencystore/storage.go (hunk 2 above)
Comment (black-adder) [inline plugin/storage/es/dependencystore/storage.go]: we have the issue here where we only want to see the dependecies over the last hour but we're returning 24+ hours worth

File: plugin/storage/integration/fixtures/queries.json (hunk 1 above)
Comment (mh-park) [inline plugin/storage/integration/fixtures/queries.json]: https://github.com/uber/jaeger/blob/es-search-ignore-indices/plugin/storage/es/spanstore/reader.go#L120

File: plugin/storage/es/dependencystore/storage.go (hunk 2 above)
Comment (mh-park) [inline plugin/storage/es/dependencystore/storage.go]: well, is it even necessary if the lookback falls on the same day?

Comment (mh-park) [inline plugin/storage/es/dependencystore/storage.go]: no this just collects the indices that we're going to query over. it'll only retrieve the ones over the last hour because the query is a range query. the query is on 
https://github.com/uber/jaeger/blob/es-search-ignore-indices/plugin/storage/es/dependencystore/storage.go#L101
https://github.com/uber/jaeger/blob/es-search-ignore-indices/plugin/storage/es/dependencystore/storage.go#L121

Comment (black-adder) [inline plugin/storage/es/dependencystore/storage.go]: right, but we're still telling the query service to search in the previous index even though we don't have to

File: plugin/storage/es/dependencystore/storage.go [hunk 3]
```diff
@@ -124,13 +124,15 @@ func buildTSQuery(endTs time.Time, lookback time.Duration) elastic.Query {
...
-	indices = append(indices, indexName(ts))
-	for lookback > 0 {
+	firstIndex := indexName(ts.Add(-lookback))
+	currentIndex := indexName(ts)
+	for currentIndex != firstIndex {
+		indices = append(indices, currentIndex)
 		ts = ts.Add(-24 * time.Hour)
-		indices = append(indices, indexName(ts))
+		currentIndex = indexName(ts)
 		lookback -= 24 * time.Hour
```
Comment (black-adder) [inline plugin/storage/es/dependencystore/storage.go]: dont need
